tests/components/amberelectric/conftest.py

```diff
@@ -13,3 +59,103 @@ def mock_setup_entry() -> Generator[AsyncMock]:
         "homeassistant.components.amberelectric.async_setup_entry", return_value=True
     ) as mock_setup_entry:
         yield mock_setup_entry
+
+
+@pytest.fixture
+async def setup_general(hass: HomeAssistant) -> AsyncGenerator[Mock]:
+    """Set up general channel."""
+    MockConfigEntry(
+        domain="amberelectric",
+        data={
+            CONF_SITE_NAME: "mock_title",
+            CONF_API_TOKEN: MOCK_API_TOKEN,
+            CONF_SITE_ID: GENERAL_ONLY_SITE_ID,
+        },
+    ).add_to_hass(hass)
+
+    instance = Mock()
+    with patch(
+        "amberelectric.AmberApi",
+        return_value=instance,
+    ) as mock_update:
+        instance.get_current_prices = Mock(return_value=GENERAL_CHANNEL)
+        assert await async_setup_component(hass, DOMAIN, {})
```

Comment:
let's not use async_setup_component, rather use `hass.config_entries.async_setup(entry.entry_id)`